src/Items/ItemHandler.cpp

```diff
@@ -63,16 +63,15 @@
 
 
 
-bool cItemHandler::m_HandlerInitialized = false;
-cItemHandler * cItemHandler::m_ItemHandler[2268];
+std::array<std::vector<std::unique_ptr<cItemHandler>>, E_ITEM_LAST + 1> cItemHandler::m_ItemHandler;
```

Comment:
Yeah, I also thought of this. But I think both versions have their advantages/disadvantages.
The "problem" with passing the datavalues to the function is, that you have such constructs:
> Item.getHandle().EatFood(Item.m_Damage);

This looks silly in my eyes, so that you would need to implement all cItemHandler functions into cItem to remove passing already avaible data.

If the vectors seem a to high performance implication, we could use unique_ptr<cItemHandler[]> and implement something to track max ItemData values...
But I think the performance shouldn't get worse, because we do only allocate memory 1-4 times, when the ItemHandlers are created and pushed.